src/scores/categorical/contingency_impl.py

```diff
@@ -234,6 +279,142 @@ def specificity(self):
         s = cd["tn_count"] / (cd["tn_count"] + cd["fp_count"])
         return s
 
+    def recall(self):
+        """
+        Identical to probability of detection.
+
+        https://en.wikipedia.org/wiki/Precision_and_recall
+        """
+        return self.probability_of_detection()
+
+    def precision(self):
+        """
+        Identical to the Success Ratio.
+
+        https://en.wikipedia.org/wiki/Precision_and_recall
+        """
+        return self.success_ratio()
+
+    def f1_score(self):
+        """
+        Calculates the F1 score.
+        https://en.wikipedia.org/wiki/F-score
+        """
+        cd = self.counts
+        f1 = 2 * cd["tp_count"] / (2 * cd["tp_count"] + cd["fp_count"] + cd["fn_count"])
+        return f1
+
+    def equitable_threat_score(self):
+        """
+        Calculates the Equitable threat score (also known as the Gilbert skill score).
+
+        How well did the forecast "yes" events correspond to the observed "yes"
+        events (accounting for hits due to chance)?
+
+        Range: -1/3 to 1, 0 indicates no skill. Perfect score: 1.
+
+        Hogan, R.J., Ferro, C.A., Jolliffe, I.T. and Stephenson, D.B., 2010.
+        Equitability revisited: Why the “equitable threat score” is not equitable.
+        Weather and Forecasting, 25(2), pp.710-726.
+        """
+        cd = self.counts
+        hits_random = (cd["tp_count"] + cd["fn_count"]) * (cd["tp_count"] + cd["fp_count"]) / cd["total_count"]
+        ets = (cd["tp_count"] - hits_random) / (cd["tp_count"] + cd["fn_count"] + cd["fp_count"] - hits_random)
+
+        return ets
+
+    def gilberts_skill_score(self):
+        """
+        Calculates the Gilbert skill score (also known as the Equitable threat score).
+
+        How well did the forecast "yes" events correspond to the observed "yes"
+        events (accounting for hits due to chance)?
+
+        Range: -1/3 to 1, 0 indicates no skill. Perfect score: 1.
+
+        Hogan, R.J., Ferro, C.A., Jolliffe, I.T. and Stephenson, D.B., 2010.
+        Equitability revisited: Why the “equitable threat score” is not equitable.
```

Comment:
I think the original citation might be Gilbert, G. K., 1884: Finley’s tornado predictions. Amer. Meteor. J., 1 , 166–172. We should try to access this through the library and if that's correct provide the original citation.

Reply:
I have raised an issue to investigate this post-merge. The provided reference does provide a mathematical definition of the skill and includes a lot of relevant commentary.